src/libpriv/rpmostree-scripts.c

```diff
@@ -957,6 +958,21 @@ rpmostree_deployment_sanitycheck (int           rootfs_fd,
   rpmostree_bwrap_append_child_argv (bwrap, "/usr/bin/true", NULL);
   if (!rpmostree_bwrap_run (bwrap, cancellable, error))
     return FALSE;
-  sd_journal_print (LOG_INFO, "sanitycheck(/usr/bin/true) successful");
+
+  /* Also check that we can load the rpmdb.
+   * See https://github.com/projectatomic/rpm-ostree/issues/1566. */
+  g_autoptr(RpmOstreeRefSack) sack = rpmostree_get_refsack_for_root (rootfs_fd, ".", error);
```

Comment:
I am not sure if this truly works well today but I'd like to support anyone who wants to use rpm-ostree client-side to manage "pure ostree" commits generated by something else. 

One way we could handle this is to only perform this check if the deployment is layered.

Reply:
Ahh, interesting. Though... why would one want to use rpm-ostree vs ostree in that case? I mean, there's definitely a few added value items in rpm-ostree that are not RPM-related (e.g. `initramfs`), though the other non-RPM ones are already mostly covered by `ostree admin`. I don't mind tweaking this, though ISTM it'd be more valuable for now to keep that sanity check even on base layers until that use case shows up?

Reply:
One is the DBus API, which e.g. Cockpit can talk to.

Reply:
There are almost certainly other things that don't work with this today (e.g. I bet us simply trying to gather package diffs would fail), and I don't think we have any tests for it - so feel free to ignore this for now.